### PR TITLE
HPCC-14636 Add parallel quick sorts to roxie (for comparison)

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review

File: common/thorhelper/roxiehelper.hpp [hunk 1]
```diff
@@ -104,6 +104,7 @@ typedef enum { heapSortAlgorithm, insertionSortAlgorithm,
               mergeSortAlgorithm, spillingMergeSortAlgorithm,
               parallelMergeSortAlgorithm, spillingParallelMergeSortAlgorithm,
               tbbQuickSortAlgorithm, tbbStableQuickSortAlgorithm,
```
Comment (richardkchapman) [inline common/thorhelper/roxiehelper.hpp]: Are tbbQuickSort and tbbStableQuickSort not parallel?

Perhaps some comments on this enum describing what is what would be helpful?

What algo is used if they just indicate they want a parallel sort without being specific about which one (can they do that?)

**Comment by richardkchapman** [comment]
@ghalliday Couple of questions

**Comment by ghalliday** [comment]
@richardkchapman updated to reflect comments.  I also changed the spill logic to allow it to be set to false.
I added a couple of jiras to cover cleaning up parallel/stable/spilling options.

**Pull request merged**